### PR TITLE
Add ability to proxy via NGINX

### PR DESCRIPTION
**Pull request by kierenevans**
Useful for SSL offloading, for instance.

### CODE REVIEW

File: templates/default/nginx_site.conf.erb [hunk 1]
```diff
@@ -48,6 +48,17 @@ server {
       case data['mode']
       when 'restricted'
             %>deny all;<%
+      when 'proxy'
+            %>
+            proxy_pass <%= data['proxy']['location'] %>;
+            proxy_set_header Host $host;
+            proxy_set_header X-Forwarded-For $remote_addr;
+            <% if @protocol == 'https' %>proxy_set_header X-Forwarded-Proto https;<% end %>
```
Comment (andytson-inviqa) [inline templates/default/nginx_site.conf.erb]: could this maybe be
```
proxy_set_header X-Forwarded-Proto $scheme;
```

**Comment by andytson-inviqa** [comment]
Feels a bit too much of a hack to be re-using something rendering the fastcgi and other locations for them to get ignored because the proxy location takes precedence

**Comment by kierenevans** [comment]
Do we need an implicit turning off of the php_support flag, or would a separate vhost be better?

**Comment by andytson-inviqa** [comment]
I'm thinking perhaps migrating the endpoint/php_support style to a default location attribute, allowing a simple mode change to make it do a proxy instead of the endpoint/php_support style.

I'm also thinking changing the fastcgi/proxy to [at]handlers using a method @shrikeh and I have been discussing

**Pull request merged**